getter.go

```diff
@@ -132,11 +138,15 @@ func (g *getter) queueFile(url *url.URL) (http.Header, error) {
 
 	g.chunkWg.Add(1)
 	resp, err := g.retryRequest("GET", url.String(), nil)
+
+	// resp could be nil, depending on the error.
 	if err != nil {
-		return resp.Header, err
+		logger.debugPrintf("ERROR on queueFile", errgo.Mask(err))
+		return nil, err
 	}
 
```

Comment:
This change actually avoids accessing resp (removing line 136) when err is not nil (I believe we had crashed here before). I can add more checking to return resp.header when resp is not nil, return nil otherwise.